app/src/main/java/com/github/tonytangandroid/daggertutorial/dagger/MainActivityModule.java

```diff
@@ -1,10 +1,20 @@
 package com.github.tonytangandroid.daggertutorial.dagger;
 
+import android.app.Activity;
+import com.github.tonytangandroid.daggertutorial.MainActivity;
+import com.github.tonytangandroid.daggertutorial.MainView;
 import dagger.Module;
+import dagger.Provides;
 import dagger.hilt.InstallIn;
 import dagger.hilt.android.components.ActivityComponent;
 
 /** Created by ztang on 2/8/18. */
 @Module
 @InstallIn(ActivityComponent.class)
-abstract class MainActivityModule {}
+abstract class MainActivityModule {
+
+  @Provides
+  static MainView provideContext(Activity activity) {
+    return ((MainActivity) activity);
```

Comment:
`ActivityComponent` will provide `Activity`, which makes this hacky possible. But this hack definitely could  be resolved through Dagger Hilt Android compiler.

Reply:
This hack is really bad. This just works by accidentally. If you introduce a  second `Activity`, the app will crash because it could not be casted to `MainActivity`. : (